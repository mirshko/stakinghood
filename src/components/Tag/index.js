import React from "react";
import { Box as RebassBox } from "rebass";
import styled from "styled-components";

const Box = styled(RebassBox)`
  align-items: center;
  border-radius: 5px;
  display: inline-flex;
  text-transform: uppercase;
  letter-spacing: 0.64px;
  font-weight: var(--weight-medium);
  height: 24px;
  padding: 1px 8px 0;
`;

const Tag = props => (
  <Box bg="var(--cosmos-brand)" color="white" {...props} fontSize={2}>
    {props.children}
  </Box>
);

export default Tag;
