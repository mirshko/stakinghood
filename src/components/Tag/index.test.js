import React from "react";
import { shallow } from "enzyme";

import Tag from "./index";

describe("<Tag />", () => {
  it("it works", () => {
    const wrapper = shallow(<Tag />);
    expect(wrapper).toMatchSnapshot();
  });
});
