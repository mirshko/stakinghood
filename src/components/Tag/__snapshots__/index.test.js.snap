// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> it works 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htoDjs",
        "isStatic": false,
        "rules": Array [
          "box-sizing: border-box;",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          "
  align-items: center;
  border-radius: 5px;
  display: inline-flex;
  text-transform: uppercase;
  letter-spacing: 0.64px;
  font-size: 16px;
  background-color: var(--brand);
  font-weight: var(--weight-medium);
  height: ",
          [Function],
          ";
  padding: 1px 8px 0;
  color: #fff;
",
        ],
      },
      "displayName": "Styled(Styled(styled.div))",
      "foldedComponentIds": Array [
        "sc-bdVaJa",
        "sc-htpNat",
      ],
      "propTypes": Object {
        "fontFamily": [Function],
        "fontWeight": [Function],
        "letterSpacing": [Function],
        "lineHeight": [Function],
        "textAlign": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-htoDjs",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
